ocaml/typing/typecore.ml

```diff
@@ -8564,6 +8586,14 @@ and type_let_def_wrap_warnings
     end
     else exp_env
   in
+  let exp_env = if is_recursive then begin
+    (* (less-tco project) Used to infer whether function applications should be tail-call
```

Comment:
Is this a CR to be addressed before merging the branch, or a permanent documentation comment? If the former, please write in the same format as the other CRs you've left. If the latter, I think we can drop the `(less-tco project)` opening.